src/goto-checker/symex_bmc.cpp

```diff
@@ -43,15 +44,32 @@ void symex_bmct::symex_step(
   statet &state)
 {
   const source_locationt &source_location = state.source.pc->source_location;
-
+  // Turn on throwing invariants so we can safely log invariant failure during Symex
```

Comment:
:question: why not just wrap the whole thing in a try catch? Isn't that a more idiomatic way to accomplish the same thing?

Reply:
I don’t know about idiomatic, CPPCoreGuidelines recommends use of RAII ([1](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Re-catch), [2](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Re-finally)), though in this case the problem is this just doesn’t quite work with our setup because `uncaught_exceptions` is C++17 apparently. I can fix the scope guard thing (it can still be done, logic is just slightly different) or I can change this to try catch. I do have a preference for the RAII based approach but I’m not married to it if other people dislike it very much.